Objects/object.c

```diff
@@ -1190,9 +1193,12 @@ PyObject_SetAttr(PyObject *v, PyObject *name, PyObject *value)
             return -1;
         }
 
-        if(Py_CHECKWRITE(v)){
+        if (Py_CHECKWRITE(v)) {
+            if (!Py_IS_REGION_AWARE(tp)) {
+                _PyObject_mark_region_as_dirty(v);
+            }
```

Comment:
Should we modify CHECKWRITE to do this as well?

Reply:
Ah you mean `Py-CHECKWRITE(obj)` marks the region as dirty if `!Py_IS_REGION_AWARE(obj->ob_type)`?

Reply:
Yep

Reply:
That's excellent! And obvious in hindsight.

Reply:
If we add that check to CHECKWRITE that will add that check to many more places that what I think we currently would need it. Maybe that's OK because branch prediction.